### ISSUE TITLE
Strange opening of 4.0.0-beta6

### ISSUE THREAD

**Issue by borekb** [careteditor/caret#612]
Here is what happens to me on Windows:

1. I've got a folder open and one file from it active, say, `file.md`.
2. I close Caret entirely.
3. I open the app again (a click in the taskbar).
4. It flashes for a bit (~300ms) and then disappears again. Like if the application crashed or maybe started and shut down immediately.
5. I click the taskbar icon again and this time, two windows of Caret open: one with my folder and `file.md`, the other one with `Welcome.md` (the built-in document that I remember from the first launch of Caret Beta).

I can repeat this reliably. Not sure whether you'd be able to reproduce on your end; let me know if I can provide any additional info. Thanks!

**Comment by borekb** [comment]
Continuation:

6. I right-clicked taskbar and selected "Close all". This closed both windows of Caret at the same time.
7. Next launch of Caret Beta opened just my folder and `file.md`.

Since these two last steps, everything seems to be behaving correctly.

**Comment by astoilkov** [comment]
Hi @borekb,

I tried to replicate the problem with the (~300ms) flashing but without success. Did this happen only once or it happens every time?

Regarding the "Close All" scenario I can successfully replicate the problem even in the official Caret release.

**Comment by borekb** [comment]
Yes, when the Caret was in a broken state (which I'm not sure how it got to that), I was able to start Caret Beta, see it disappear, click the app icon again, see it open two windows, close them one by one, try to start Caret again to see it disappear again. It was repeatable.

**Comment by astoilkov** [comment]
Is there something else that doesn't work when in this broken state. If you understand you can get to that state please let us know.

**Comment by borekb** [comment]
All seemed OK otherwise, just the window management was weird. I'll report back if this happens again.
